
From: Rusty Russell <rusty@rustcorp.com.au>

Agustin Martin <agmartin@debian.org> pointed out that this doesn't work:

	options ide-mod options="ide=nodma hdc=cdrom"

The quotes are understood by kernel/params.c (ie.  it skips over spaces
inside them), but are not stripped before handing to the underlying
function.  They should be.


---

 25-akpm/kernel/params.c |    7 +++++++
 1 files changed, 7 insertions(+)

diff -puN kernel/params.c~strip-param-quotes kernel/params.c
--- 25/kernel/params.c~strip-param-quotes	2004-04-03 03:00:06.033314352 -0800
+++ 25-akpm/kernel/params.c	2004-04-03 03:00:06.036313896 -0800
@@ -96,6 +96,13 @@ static char *next_arg(char *args, char *
 	else {
 		args[equals] = '\0';
 		*val = args + equals + 1;
+
+		/* Don't include quotes in value. */
+		if (**val == '"') {
+			(*val)++;
+			if (args[i-1] == '"')
+				args[i-1] = '\0';
+		}
 	}
 
 	if (args[i]) {

_
